src/hardware/mouse/mouse_config.cpp

```diff
@@ -39,44 +38,53 @@ CHECK_NARROWING();
 
 // #define ENABLE_EXPLORER_MOUSE
 
-MouseConfig mouse_config;
+MouseConfig     mouse_config;
 MousePredefined mouse_predefined;
 
-static struct {
-	const std::string param_standard     = "standard";
-	const std::string param_intellimouse = "intellimouse";
+static const std::string capture_type_seamless_str  = "seamless";
+static const std::string capture_type_onclick_str   = "onclick";
+static const std::string capture_type_onstart_str   = "onstart";
+static const std::string capture_type_nomouse_str   = "nomouse";
+
+static const std::string model_ps2_standard_str     = "standard";
+static const std::string model_ps2_intellimouse_str = "intellimouse";
 #ifdef ENABLE_EXPLORER_MOUSE
-	const std::string param_explorer = "explorer";
+static const std::string model_ps2_explorer_str     = "explorer";	
 #endif
-} models_ps2;
-
-static struct {
-	const std::string param_2button     = "2button";
-	const std::string param_3button     = "3button";
-	const std::string param_wheel       = "wheel";
-	const std::string param_msm         = "msm";
-	const std::string param_2button_msm = "2button+msm";
-	const std::string param_3button_msm = "3button+msm";
-	const std::string param_wheel_msm   = "wheel+msm";
-} models_com;
+
+static const std::string model_com_2button_str      = "2button";
+static const std::string model_com_3button_str      = "3button";
+static const std::string model_com_wheel_str        = "wheel";
+static const std::string model_com_msm_str          = "msm";
+static const std::string model_com_2button_msm_str  = "2button+msm";
+static const std::string model_com_3button_msm_str  = "3button+msm";
+static const std::string model_com_wheel_msm_str    = "wheel+msm";
+
+static const char *list_capture_types[] = {
+	capture_type_seamless_str.c_str(),
+	capture_type_onclick_str.c_str(),
+	capture_type_onstart_str.c_str(),
+	capture_type_nomouse_str.c_str(),
+	nullptr
+};
```

Comment:
Now I think we can go further :-); the `std::string`'s can be collapsed down to compile-time char arrays:

```diff
diff --git a/src/hardware/mouse/mouse_config.cpp b/src/hardware/mouse/mouse_config.cpp
index 5da23b3a2..cce3093b2 100644
--- a/src/hardware/mouse/mouse_config.cpp
+++ b/src/hardware/mouse/mouse_config.cpp
@@ -41,51 +41,51 @@ CHECK_NARROWING();
 MouseConfig     mouse_config;
 MousePredefined mouse_predefined;
 
-static const std::string capture_type_seamless_str  = "seamless";
-static const std::string capture_type_onclick_str   = "onclick";
-static const std::string capture_type_onstart_str   = "onstart";
-static const std::string capture_type_nomouse_str   = "nomouse";
+constexpr auto capture_type_seamless_str  = "seamless";
+constexpr auto capture_type_onclick_str   = "onclick";
+constexpr auto capture_type_onstart_str   = "onstart";
+constexpr auto capture_type_nomouse_str   = "nomouse";
 
-static const std::string model_ps2_standard_str     = "standard";
-static const std::string model_ps2_intellimouse_str = "intellimouse";
+constexpr auto model_ps2_standard_str     = "standard";
+constexpr auto model_ps2_intellimouse_str = "intellimouse";
 #ifdef ENABLE_EXPLORER_MOUSE
-static const std::string model_ps2_explorer_str     = "explorer";      
+constexpr auto model_ps2_explorer_str     = "explorer";
 #endif
 
-static const std::string model_com_2button_str      = "2button";
-static const std::string model_com_3button_str      = "3button";
-static const std::string model_com_wheel_str        = "wheel";
-static const std::string model_com_msm_str          = "msm";
-static const std::string model_com_2button_msm_str  = "2button+msm";
-static const std::string model_com_3button_msm_str  = "3button+msm";
-static const std::string model_com_wheel_msm_str    = "wheel+msm";
-
-static const char *list_capture_types[] = {
-       capture_type_seamless_str.c_str(),
-       capture_type_onclick_str.c_str(),
-       capture_type_onstart_str.c_str(),
-       capture_type_nomouse_str.c_str(),
-       nullptr
+constexpr auto model_com_2button_str     = "2button";
+constexpr auto model_com_3button_str     = "3button";
+constexpr auto model_com_wheel_str       = "wheel";
+constexpr auto model_com_msm_str         = "msm";
+constexpr auto model_com_2button_msm_str = "2button+msm";
+constexpr auto model_com_3button_msm_str = "3button+msm";
+constexpr auto model_com_wheel_msm_str   = "wheel+msm";
+
+constexpr const char *list_capture_types[] = {
+        capture_type_seamless_str,
+        capture_type_onclick_str,
+        capture_type_onstart_str,
+        capture_type_nomouse_str,
+        nullptr,
 };
 
-static const char *list_models_ps2[] = {
-       model_ps2_standard_str.c_str(),
-       model_ps2_intellimouse_str.c_str(),
+constexpr const char *list_models_ps2[] = {
+        model_ps2_standard_str,
+        model_ps2_intellimouse_str,
 #ifdef ENABLE_EXPLORER_MOUSE
-       model_ps2_explorer_str.c_str(),
+        model_ps2_explorer_str,
 #endif
-       nullptr
+        nullptr,
 };
 
-static const char *list_models_com[] = {
-       model_com_2button_str.c_str(),
-       model_com_3button_str.c_str(),
-       model_com_wheel_str.c_str(),
-       model_com_msm_str.c_str(),
-       model_com_2button_msm_str.c_str(),
-       model_com_3button_msm_str.c_str(),
-       model_com_wheel_msm_str.c_str(),
-       nullptr
+constexpr const char *list_models_com[] = {
+        model_com_2button_str,
+        model_com_3button_str,
+        model_com_wheel_str,
+        model_com_msm_str,
+        model_com_2button_msm_str,
+        model_com_3button_msm_str,
+        model_com_wheel_msm_str,
+        nullptr,
 };
 
 static const std::vector<uint16_t> list_rates = {
@@ -257,8 +257,7 @@ static void config_init(Section_prop &secprop)
 
        // General configuration
 
-       prop_str = secprop.Add_string("mouse_capture", always,
-                                     capture_type_onstart_str.c_str());
+       prop_str = secprop.Add_string("mouse_capture", always, capture_type_onstart_str);
        assert(prop_str);
        prop_str->Set_values(list_capture_types);
        prop_str->Set_help(
@@ -319,8 +318,9 @@ static void config_init(Section_prop &secprop)
 
        // Physical mice configuration
 
-       prop_str = secprop.Add_string("ps2_mouse_model", only_at_start,
-                                     model_ps2_intellimouse_str.c_str());
+       prop_str = secprop.Add_string("ps2_mouse_model",
+                                     only_at_start,
+                                     model_ps2_intellimouse_str);
        assert(prop_str);
        prop_str->Set_values(list_models_ps2);
        prop_str->Set_help(
@@ -333,8 +333,9 @@ static void config_init(Section_prop &secprop)
 #endif
                "Default: intellimouse");
 
-       prop_str = secprop.Add_string("com_mouse_model", only_at_start,
-                                     model_com_wheel_msm_str.c_str());
+       prop_str = secprop.Add_string("com_mouse_model",
+                                     only_at_start,
+                                     model_com_wheel_msm_str);
        assert(prop_str);
        prop_str->Set_values(list_models_com);
```

Reply:
OK, changed.